File: Python_Trainings/convertNumber.py
# -*- Mode: Python; coding: utf-8; indent-tabs-mpythoode: nil; tab-width: 4 -*-

# CODEWARS
# https://www.codewars.com/kata/5583090cbe83f4fd8c000051
#
# "Convert number to reversed array of digits"


def digitize(n):

    if n == 0:
        return [0]
    if n < 0:
        n *= -1
    
    arr = []
    while n > 0:
        arr += [n % 10]
        n /= 10
    return arr

def testEqual(test, str1, str2):
    print(test, str1 == str2)

def main():
    testEqual(1, digitize(35231),[1,3,2,5,3])
    testEqual(2, digitize(348597),[7,9,5,8,4,3])
    testEqual(3, digitize(0),[0])
    testEqual(4, digitize(1),[1])
    testEqual(5, digitize(10),[0,1])
    testEqual(6, digitize(100),[0,0,1])
    testEqual(7, digitize(-1000),[0,0,0,1])

if __name__ == '__main__':
    main()

File: CSharp_Trainings/convertNumber.cs
/*
CODEWARS
https://www.codewars.com/kata/5583090cbe83f4fd8c000051

"Convert number to reversed array of digits"
*/

using System;
using System.Collections.Generic;

namespace Solution
{
  class Digitizer
  {
    public static long[] Digitize(long n)
    {
      List<long> vectorOut = new List<long>();
      
      if (n< 0) n *= -1;
      while ( n >= 10) {
        vectorOut.Add(n % 10);
        n /= 10;
      }
      vectorOut.Add(n);

      long[] output = vectorOut.ToArray();

      return output;
    }
  }

  public class SolutionTest
  {
    public static bool testEqual(long[] v1, long[] v2)
    {
      if (v1.Length != v2.Length) return false;
      for (long i = 0; i < v1.Length; ++i) {
        if (v1[i] != v2[i]) return false;
      }
      return true;
    }

    public static void Main(string[] args)
    {
      Console.Write(testEqual(
        new long[] {1, 3, 2, 5, 3},Digitizer.Digitize(35231))
      );
    }
  }
  
}